plugins/arElasticSearchPlugin/lib/model/arElasticSearchFunction.class.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class arElasticSearchFunction extends arElasticSearchModelBase
+{
+  public function populate()
+  {
+    $errors = array();
+
+    $sql  = 'SELECT function.id';
+    $sql .= ' FROM '.QubitFunction::TABLE_NAME.' function';
+    $sql .= ' JOIN '.QubitObject::TABLE_NAME.' object ON function.id = object.id';
+    $sql .= ' WHERE object.class_name = ?';
+    $sql .= ' ORDER BY function.lft';
+
+    $actors = QubitPdo::fetchAll($sql, array('QubitFunction'));
```

Comment:
I've never realized we use fetchAll here (in similar classes for other entities). The only reason why we do it seems to be that `count()`. It sounds like we should use SQL instead for that count and then make another query using just `fetch` so we can use a scrollable cursor in the loop to save memory right? It doesn't have to happen now though unless you have time :)